### PR TITLE
detect/msg: convert to FAIL/PASS API

### PR DESCRIPTION
**Pull request by jufajardini**
Issue 4053. Adjust code formatting style (wrap long lines).
Replace SigInit with DetectEngineAppendSig.

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4053

Link to previous PR:
https://github.com/OISF/suricata/pull/5505

Describe changes:
- Wrap long lines to keep code complaint with formatting guidelines
- Replace SigInit with DetectEngineAppendSig
- Remove no longer necessary SigFree

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by jufajardini** [comment]
Hi! Can you please let me know if there is anything else I must do for this PR to be accepted?

Thanks a lot for your time. :)

**Review by victorjulien — APPROVED** [review]
Looks good, thanks for your contribution!

**Comment by jufajardini** [comment]
> Looks good, thanks for your contribution!

Nice, thanks for your feedback, so I could make it! One quesiton: should I change redmine issue's status to resolved or something, or do I leave it as "In Review", still?

**Comment by catenacyber** [comment]
> should I change redmine issue's status to "Resolved" or something, or do I leave it as "In Review", still?

Please wait until the PR is merged :-)

File: src/detect-msg.c
Comment (jufajardini) [inline src/detect-msg.c:145]: I was working on a similar issue (for detect/iprep) today, and discussing some doubts with a friend, and we got confused regarding the clean up stage of these tests, once we change from the goto end style to the FAIL/PASS macros...

From what we could see, once we use the FAILs, the method will return immediately, skipping any code related to freeing resources. 

I tried to investigate further a bit, but couldn't tell nor find if that is going to happen elsewhere, when we have the macros API. So I am wondering if that is expected behaviour, because these non-freed resources aren't critical; or if this is a change to be implemented later; or maybe I missed something...

Commenting here, because depending on the answers, there will be changes in this one, as well...

Comment (inashivb) [reply src/detect-msg.c:145]: You are correct. We do not have to worry about freeing resources in case of failed tests. We just need to make sure though that in case of success, we free all the resources.

Comment (jufajardini) [reply src/detect-msg.c:145]: I see! Thank you. Is that because most tests are expected to pass, so we don't have to worry about memory leaks and the likeṣ?

**Comment by victorjulien** [comment]
Merged as part of #5646, thanks a lot for your contribution!

**Pull request closed without merging**
